tests/ReadPropertiesTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace DigipolisGent\Tests\Robo\Task\General;
+
+use League\Container\ContainerAwareInterface;
+use League\Container\ContainerAwareTrait;
+use Robo\Contract\ConfigAwareInterface;
+use Robo\Common\CommandArguments;
+use Robo\Robo;
+use Robo\TaskAccessor;
+use Symfony\Component\Console\Output\NullOutput;
+
+class ReadPropertiesTest extends \PHPUnit_Framework_TestCase implements ContainerAwareInterface, ConfigAwareInterface
+{
+
+    use \DigipolisGent\Robo\Task\General\loadTasks;
+    use TaskAccessor;
+    use ContainerAwareTrait;
+    use CommandArguments;
+    use \Robo\Task\Base\loadTasks;
+    use \Robo\Common\ConfigAwareTrait;
+
+    /**
+     * Set up the Robo container so that we can create tasks in our tests.
+     */
+    public function setUp()
+    {
+        $container = Robo::createDefaultContainer(null, new NullOutput());
+        $this->setContainer($container);
+        $this->setConfig(Robo::config());
+    }
+
+
+    protected function getRandomString($length = 5)
+    {
+        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+        $charactersLength = strlen($characters);
+        $randomString = '';
+        for ($i = 0; $i < $length; $i++) {
+            $randomString .= $characters[rand(0, $charactersLength - 1)];
+        }
+        return $randomString;
+    }
+
+    /**
+     * Scaffold the collection builder.
+     *
+     * @return \Robo\Collection\CollectionBuilder
+     *   The collection builder.
+     */
+    public function collectionBuilder()
+    {
+        $emptyRobofile = new \Robo\Tasks();
+
+        return $this->getContainer()
+            ->get('collectionBuilder', [$emptyRobofile]);
+    }
+
+    public function testRun() {
+        // Test with root config and overrides.
+        $this->getConfig()->set('digipolis.root.project', __DIR__ . '/root');
+        $result = $this->taskReadProperties(__DIR__ . '/web')
```

Comment:
They can be anywhere, that's why it's an argument so devs can choose which folder to scan. I just called it 'web' for the sake of it, had to have a folder to test. The only requirement right now is as you say, you need a RoboFile.php. An other option would be to search for less general properties files (robo.properties.yml and robo.default.properties.yml or digipolis.propteries.yml and digipolis.default.properties.yml).